x/schema/keeper/schema.go

```diff
@@ -0,0 +1,94 @@
+package keeper
+
+import (
+	"encoding/binary"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/google/uuid"
+	"github.com/sonr-io/sonr/x/schema/types"
+)
+
+func (k Keeper) GenerateKeyForDID() string {
+	id := uuid.New()
+
+	return id.String()
+}
+
+func (k Keeper) GetWhatIsCount(ctx sdk.Context) uint64 {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), []byte{})
+	byteKey := types.KeyPrefix(types.SchemaCountKey)
+	bz := store.Get(byteKey)
+
+	if bz == nil {
+		return 0
+	}
+
+	return binary.BigEndian.Uint64(bz)
+}
+
+// SetWhoIsCount set the total number of whoIs
+func (k Keeper) SetSchemaCount(ctx sdk.Context, count uint64) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), []byte{})
+	byteKey := types.KeyPrefix(types.SchemaCountKey)
+	bz := make([]byte, 8)
+	binary.BigEndian.PutUint64(bz, count)
+	store.Set(byteKey, bz)
+}
+
+// GetSchemaFromCreator returns a WhoIs whos DIDDocument contains the given controller
+func (k Keeper) GetWhatIsFromCreator(ctx sdk.Context, creator string) (val []types.WhatIs, found bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.SchemaKeyPrefix))
+	iterator := sdk.KVStorePrefixIterator(store, []byte{})
+	defer iterator.Close()
+	var vals []types.WhatIs = make([]types.WhatIs, 0)
+	for ; iterator.Valid(); iterator.Next() {
+		var instance types.WhatIs
+		k.cdc.MustUnmarshal(iterator.Value(), &instance)
```

Comment:
I don't think we should panic if this fails. We won't be able to return a valid error response